load_testing_locust.md

```diff
@@ -0,0 +1,592 @@
+HTTPBIN="http://httpbin-httpbin.apps.fwaters-ccs2.xqpg.s2.devshift.org/nothing/10"
+
+ACCESS_TOKEN=`curl \
+  -d "client_id=$CLIENT_ID" -d "client_secret=$CLIENT_SECRET" \
+  -d "username=$UNAME" -d "password=$PASSWORD" \
+  -d "grant_type=password" \
+  "$KCHOST/auth/realms/$REALM/protocol/openid-connect/token"  | jq -r '.access_token'`
+echo $ACCESS_TOKEN
+echo THREESCALE_ROUTE ===============================================================
+curl -X POST -H "Authorization: Bearer $ACCESS_TOKEN" $THREESCALE_ROUTE -d '{"login":"my_login","password":"my_password"}'
+echo HTTPBIN ==================================================================
+curl -X POST $HTTPBIN -d '{"login":"my_login","password":"my_password"}'
+```
+
+
+Once this is confirmed you can populate the required auth file. See Auth section below.
+
+<br>
+
+## Setting up locust
+
+Locust is a load testing tool that can be run locally or on a container on another cluster
+
+* Fork repo - [https://github.com/integr8ly/locust-integreatly-operator](https://github.com/integr8ly/locust-integreatly-operator)
+* Clone it locally
+* Install locust
+
+```
+pip3 install locust 
+```
+
+**Config**
+
+constant_pacing is used to get to the correct number of requests per second(rps) for the load testing on locust. It also takes into account the number of users. It is the inverse of constant_throughput. In order to increase the rps you need to decrease the constant pacing value. When changing this be sure to only update in small increments. More info on this function can be found [here](https://docs.locust.io/en/stable/api.html#locust.wait_time.constant_pacing).
+
+
+* 1.157 rps pacing value 9.25 for 100k at 11 users
+* 11.57 rps pacing value 1.25 for 1m at 14 users
+* 57.87 rps pacing value 0.207 for 5m at 12 users
+* 110.00 rps pacing value 0.09 for 10m at 10 user
+* 231.48 rps pacing value * for 20m at 12 user
+* ~550 rps pacing value 0.01 for 50m at 32 users
+* 1157.41 rps pacing value * for 100m at 12 user
+
+
+<img src="images/calculate-rps.png" width="30%" height="30%">
+
+
+
+Alternatively you can set the constant_pacing value to 1 and adjust the number of users accordingly. Example - For the 20 Million quota, that equates to 232 users and 232 requests per second.
+
+<br>
+
+## Auth file
+Locust requires an auth file. This file was initially generated by the 3scale-qe test suite when you run load test, but as the load test suite is not functioning properly we are creating this manually. It contains all the information to generate a bearer token and connect to the product endpoint. This file should be placed in the Locust directory in the https://github.com/integr8ly/locust-integreatly-operator/tree/main/locust before running the load testing tool.
+
+TOML is the recommended file format, but JSON or CSV can also be used. See examples below:
+
+## TOML
+
+<img src="images/toml-example.png" width="30%" height="30%">
+
+
+## JSON
+<img src="images/json-example.png" width="30%" height="30%">
```

Comment:
Again with the json
```suggestion
```json
{
  "host": "localhost:8000",
  "sso": "localhost:8000",
  "endpoint": "/auth/realms/protocol/openid-connect/token",
  "grant_type": "password",
  "client_id": "676d9abf",
  "client_secret": "abc123",
  "username": "testUser",
  "password": "testUser"
}
```